subsys/net/openthread/rpc/client/ot_rpc_diag.c

```diff
@@ -27,54 +27,43 @@ static otMeshLocalPrefix mesh_prefix;
 
 static void get_uint_t(int id, void *result, size_t result_size)
 {
-	const size_t cbor_buffer_size = 0;
 	struct nrf_rpc_cbor_ctx ctx;
+	uint32_t value;
 
-	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 0);
 
 	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, id, &ctx);
 
-	if (!zcbor_uint_decode(ctx.zs, result, result_size)) {
-		nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+	value = nrf_rpc_decode_uint(&ctx);
+	if (!nrf_rpc_decoding_done_and_check(&ot_group, &ctx)) {
 		ot_rpc_report_decoding_error(id);
 		return;
 	}
 
-	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+	memcpy(result, &value, result_size);
```

Comment:
Do you suggest to revert this change back?

Reply:
Well, I'm not sure how much we care about endianness but if e.g. `result_size` is 2, and you copy the first 2 bytes of `uint32_t` to the `result`, it will only work if the 2 bytes contain the least significant bytes, not the most.

We can leave it as is for now but eventually, I think we should think about improving the nrf_rpc_serialize.c helpers to validate the size of the integer being decoded so that the decoder doesn't always decode a value to `uint32_t` and then implicitly casts it to `uint16_t` or `uint8_t`. `zcbor` has its nice API that allows to pass the destination integer size and using `nrf_rpc_decode_uint` completely bypasses this size check.

Reply:
Hm... cbor doesn't send extra bytes and tries to pack digit in minimum possible number of bytes. I mean:
0x01 -> 01
0x0001 -> 01
0x00000001 -> 01
Despite the initial types for three cases are different, the final cbor data frame will be the same.
There are no any extra bytes those might be copied in wrong place. Every cbor byte is meaningful.
Furthermore, zcbor always dynamically extracts meaningful number of bytes those have to be copied. 
In advance, it sets all zeroes in provided memory:
https://github.com/NordicSemiconductor/zcbor/blob/main/src/zcbor_decode.c#L170
and after this just copies meaningful bytes considering platform endianness:
https://github.com/NordicSemiconductor/zcbor/blob/main/src/zcbor_decode.c#L162-L163
https://github.com/NordicSemiconductor/zcbor/blob/main/src/zcbor_decode.c#L175

I suspect that doing memcpy of final result is safe and endiannes has been already considered.

Reply:
Yes, I'm aware of the CBOR encoding but I'm talking about a different thing - you're not `memcpy`ing CBOR representation of an integer but `uint32_t` itself, which assumes a specific endianness.

Reply:
But nevermind, it's quite unlikely that we need to run this on big-endian anytime soon :)

Reply:
yes, I do. ZCBOR prepares to me `uint32_t` variable according to requested type length. Despite on came CBOR data this function always gets `uint32_t` back. Further, it operates with it accordingly.

Reply:
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/nrf_rpc/nrf_rpc_serialize.c#L184-L193
we always request `uint32_t` and get it back.

Reply:
I feel I'm not understood well, so let me rephrase:

1. On big-endian platform, if the CBOR contains `01`, the nrf_rpc_decode_uint() returns `0x01`, which in memory is represented as `00 00 00 01`. Thus if you then `memcpy` this value to `uint8_t` (cast to `void*`) with `result_size == sizeof(uint8_t) == 1`, you copy `00` instead of `01`, which you intended. That's why I mentioned it assumes little-endian.
2. `uint8_t some_var = nrf_rpc_decode_uint()` does not do a proper size-check. That is, if a sender was updated to send `uint16_t` in this field, and it would send `0xdead`, the receiver would first decode the value in zcbor as `(uint32_t)0xdead`, and then it would cast the value implicitly to `uint8_t`. Hence `some_var` would end up with `0xad` without raising an error. This may introduce some tricky bugs, so using `zcbor_decode_uint(..., &some_var, sizeof(some_var))` is safer because it validates that CBOR-encoded data is not larger than `some_var` that is expected to get the result of the decoding.